trustmanager/x509utils.go

```diff
@@ -507,18 +505,18 @@ func X509PublicKeyID(certPubKey data.PublicKey) (string, error) {
 	if err != nil {
 		return "", err
 	}
-	var finalAlgorithm data.KeyAlgorithm
+	pubKeyBytes, err := x509.MarshalPKIXPublicKey(cert.PublicKey)
+	if err != nil {
+		return "", err
+	}
 
+	var key data.PublicKey
 	switch certPubKey.Algorithm() {
 	case data.ECDSAx509Key:
-		finalAlgorithm = data.ECDSAKey
+		key = data.NewECDSAPublicKey(pubKeyBytes)
```

Comment:
Likewise the public key doesn't have to match the private key when constructing a private key, but I wasn't sure if this could cause problems.